Adds option to pass probe file to loadMetaData

vncntprvst opened the pull request:
Also adds probe file for Cambridge Neurotech P series Chronic 32 channel probes

Commit:
Adds option to pass probe file to loadMetaData

Also adds probe file for Cambridge Neurotech P series Chronic 32 channel probes

Commit:
Modifies bootstrap to allow for silent .prm file generation

Adds two optional fields in .meta file  'paramDlg' and 'advancedParam'. Also trusts the meta file for the .prb file

aliddell commented:
Looks good. I'll give it a spin when I get into the office later today and try to break it.

aliddell commented on @JRC/bootstrap.m:
```diff
@@ -110,18 +110,21 @@ function bootstrap(obj, varargin)
-        case {'Cancel', ''}
-            return;
+    if ~(exist('cfgData','var') && ... % if probe file already specified, no need to ask for it again
+            ~isempty(regexp(cfgData.probe_file,['(?<=\' filesep ')\w+?(?=.prb)'], 'once')))
```
Couple of issues with this line:

- throws if you try to bootstrap off of a .bin/.dat file instead of a .meta file
- too restrictive (requires probe file to end in `prb`)

aliddell commented:
I like the overall idea but not sure about requiring users to edit meta files which they may not have (or create them wholesale). I think arguments to the function would be better, e.g.,

`jrc bootstrap /path/to/metafile.meta /path/to/probefile.prb -noconfirm`

or

`jrc bootstrap /path/to/probefile.prb -noconfirm`

&c.

Do you agree?

Commit:
corrects cnt Pseries chronic probe geometry

vncntprvst commented:
I agree! 
I took this route because I wrote custom functions to generate meta files. But passing arguments is even simpler.

aliddell commented:
Excellent! Do you have time to take a crack at it or shall I add it to my queue?

vncntprvst commented:
I can definitely take a quick look later today and let you know.

Commit:
Fix multi-file shape bug

Commit:
Somewhat more intelligent .meta handling

- New probe files for Neuropixels 3A and 3B (staggered and aligned)
- Don't assume 3A if an Imec probe (ask!)
- Check for prb file in working directory and try to use it first

Commit:
Remove OOB spikes on import

Commit:
Merge pull request #119 from vidriotech/master

bugfixes and minor enhancements

Commit:
Include loadPath when reading in features files

Fixes #120

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Fix some issues in commit and autodelete

- autodelete commit message was incorrect
- commit fast-forward failed to account for partition

Commit:
some more bug fixes

- fixes argument not assigned when not specifying meta file
- throws an explicit error when user doesn't specify a prm file for `preview`

Commit:
Update planLoad.m

Fixes #122

Commit:
switch PCA algorithm to 'eig' in autoSplit

Commit:
add speculative operations

Commit:
revamp history mechanism

Commit:
accommodate new history structure in importv3

Commit:
remove tryImport for the hack it was

Commit:
delete 'partitionCluster'

Commit:
delete 'editSeek'

Commit:
delete 'splitCluster'

Commit:
fix small cluster bug in FigISI

Commit:
commit for TemplateClustering

Commit:
rearrange DensityPeakClustering constructor

Commit:
TemplateClustering gets revert

Commit:
remove some commented code

Commit:
remove some functions that get called in commit

Commit:
delete 'refresh'

Commit:
convert old-style history to new style

Commit:
update changelog/jrc.json

Commit:
update reassign to use new commit mechanism

Commit:
remove download link, add activity and recluster to prmCmds list

Commit:
rectification of names in autoSplit

nclack committed:
Create CODE_OF_CONDUCT.md

nclack committed:
Create LICENSE

Commit:
highlight support email

Commit:
Merge pull request #125 from JaneliaSciComp/add-code-of-conduct-1

Create CODE_OF_CONDUCT.md

Commit:
don't fail on a chunk with no detects

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
fix broken link

Commit:
fix otherFields merge in TemplateClustering

Commit:
redo features for proper splitting

Commit:
vectorize ampToProj

Commit:
FigProj split takes all spikes in polygon, not just subsetted spikes

Commit:
update changelog, jrc.json

Commit:
adds git commit hash to version info if applicable

Commit:
make allowances for Matlab 1/n

Commit:
start to update docs

Commit:
update issue template

Commit:
adds automatic inconsistent data recovery facility

Commit:
auto-recover should be opt-in

Commit:
add check for empty unitPart

Commit:
add check for empty unitPart (2/2)

Commit:
add option to export hClust to workspace

Commit:
most of output file data documented

Commit:
update example output in tutorial

agbondy committed:
autodelete: new fields to use and improved user interface

agbondy committed:
Update autoDelete.m

Commit:
Update help texts

Commit:
Merge branch 'cherry2' of git://github.com/agbondy/JRCLUST into agbondy-cherry2

Commit:
Merge branch 'agbondy-cherry2'

Commit:
update jrc.json

Commit:
support for NWB export

Commit:
Export res after a run only if exportResults is true.

Commit:
Fix hardcoded value in phy export

Commit:
Adds loadPhy to generalize import from phy-formatted sources

Commit:
kilosort import uses `loadPhy`

Commit:
SpyKING CIRCUS import supported

Commit:
add MatNWB as a submodule (and a link)

Commit:
check for degenerate feature matrices in sort

Commit:
deprecate phy-export support for Clusterings that are probably imports in the first place

Commit:
remove explicit hCfg requirement from FigWav plot

Commit:
adds "annotatedOnly" to the Clustering interface

Commit:
adds showSubset, nShown, unitIndex

Commit:
check for empty meanSiteThresh

Commit:
more compact condition in updatePlot

Commit:
showSubset and selected have setters

Commit:
showSubset

Commit:
revert hardcoded subsetting

Commit:
unit display subsetting enabled

Commit:
Fix annotation cancel bug

Commit:
support for new-style as well as old-style probe configuration in loadProbe; multishank plotting

Commit:
update changelog, jrc.json

Commit:
check for transient/dependent properties on save and load

Commit:
a more useful message when revert fails

Commit:
update CHANGELOG, jrc.json

Commit:
fix Dependent + Setter error

Commit:
save everything

Commit:
fix OOM error on reverting large history files

Commit:
fix OOB errors on delete or merge

Commit:
update CHANGELOG and jrc.json

Commit:
get aux file

Commit:
adds silent "-noconfirm" option to bootstrap

calling bootstrap with -noconfirm suppresses all dialogues

Commit:
Merge branch 'noconfirm_bootstrap'

Commit:
Merge remote-tracking branch 'origin/master'

vncntprvst commented:
Sorry it took me so long to get back to that issue. Please check the "-noconfirm" option for bootstrap. 
I also had to add a "advanced" option to specify advanced parameters. 
A probe file has been added, but feel free to disregard it.

Commit:
Merge branch 'master' into AuxChannel

Commit:
adds option to select Aux file

Commit:
Merge branch 'AuxChannel'

Commit:
fixed typo

vncntprvst commented:
This more recent commits adds the option to select the source Aux file when selecting Plot> Firing rate vs. Aux input.

vncntprvst commented:
Closing PR from master branch to generate them from dedicated branches
